docs/plugins/criteria_for_core_components.md

```diff
@@ -0,0 +1,46 @@
+# Establishing Criteria for Core vs Non-Core Components
+
+## Overview
+As OpenSearch Dashboards continues to grow, we need clear criteria to determine what should be included in core versus non-core components. This decision framework will help maintain a focused, maintainable core while allowing for robust plugin ecosystem development.
+
+## Current Challenge
+The lack of clear criteria for classifying components as core or non-core has led to:
+- Inconsistent decision-making across different features and plugins
+- Potential bloat in core components
+- Uncertainty in plugin development and maintenance strategies
+
+## Proposed Criteria for Core Components
```

Comment:
should we finalize the proposed ideas first, then list the decided criteria in docs? and the challenge can become benefits/reasoning

because if I see proposed ideas in docs, I'm not sure if I should follow them or if they are still subject to change

Reply:
Thanks, I raised PR for better communication as we can comment on a line. All the ideas are still subject to change, it is not finalized.